src/Expect.elm

```diff
@@ -234,6 +235,39 @@ atLeast =
     compareWith "Expect.atLeast" (>=)
 
 
+{-| Passes if the second and third arguments are equal within a tolerance
+specified by the first argument. This is indended to avoid failing because of
+minor inaccuracies introduced by floating point arithmetic.
+
+    -- Fails because 0.1 + 0.2 == 0.30000000000000004 (yes, seriously)
+    0.1 + 0.2 |> Expect.equal 0.3
+
+    -- So instead write this test, which passes
+    0.1 + 0.2 |> Expect.within 0.000000001 0.3
+
+
+Failures resemble code written in pipeline style, so you can tell
+which argument is which:
+
+    -- Fails because 3.14 is not close enough to pi
+    3.14 |> Expect.within 0.0001 pi
+
+    {-
+
+    3.14
+    ╷
+    │ Expect.within 0.0001
+    ╵
+    3.141592653589793
+
+    -}
+-}
+within : Float -> Float -> Float -> Expectation
+within tolerance =
+    compareWith ("Expect.within " ++ toString tolerance)
+        (\a b -> abs (a - b) < tolerance)
```

Comment:
I'd expect this to be `<=` instead of `<`

(That is, I'd expect it to be true that `0.5` is within `0.1` of `0.4`)